src/tools/schema2packager.ts

```diff
@@ -7,195 +7,63 @@
-  'unsigned', 'using', 'virtual', 'void', 'volatile', 'wchar_t', 'while', 'xor', 'xor_eq'
-];
-
-
-function generate(name: string, schema) {
-  const fields: string[] = [];
-  const api: string[] = [];
-  const clone: string[] = [];
-  const equals: string[] = [];
-  const decode: string[] = [];
-  const encode: string[] = [];
-  const toString: string[] = [];
-
-  const processValue = (field, type, typeChar, passByReference) => {
-    const [ref1, ref2] = passByReference ? ['const ', '&'] : ['', ''];
-    const fixed = (keywords.includes(field) ? '_' : '') + field;
-    const valid = `${field}_valid_`;
-
-    fields.push(`${type} ${field}_ = ${type}();`,
-                `bool ${valid} = false;`,
-                ``);
-
-    api.push(`${ref1}${type}${ref2} ${fixed}() const { return ${field}_; }`,
-             `void set_${field}(${ref1}${type}${ref2} value) { ${field}_ = value; ${valid} = true; }`,
-             `void clear_${field}() { ${field}_ = ${type}(); ${valid} = false; }`,
-             `bool has_${field}() const { return ${valid}; }`,
-             ``);
-
-    clone.push(`clone.${field}_ = entity.${field}_;`,
-               `clone.${valid} = entity.${valid};`);
-
-    equals.push(`(a.has_${field}() ? (b.has_${field}() && a.${fixed}() == b.${fixed}()) : !b.has_${field}())`);
-
-    decode.push(`} else if (name == "${field}") {`,
-                `  decoder.validate("${typeChar}");`,
-                `  decoder.decode(entity->${field}_);`,
-                `  entity->${valid} = true;`);
-
-    encode.push(`if (entity.has_${field}())`,
-                `  encoder.encode("${field}:${typeChar}", entity.${fixed}());`);
-
-    toString.push(`if (entity.has_${field}())`,
-                  `  printer.add("${field}: ", entity.${fixed}());`);
-  };
-
-  let fieldCount = 0;
-  for (const [field, descriptor] of Object.entries(schema.fields)) {
-    fieldCount++;
-    switch (typeSummary(descriptor)) {
-      case 'schema-primitive:Text':
-        processValue(field, 'std::string', 'T', true);
-        break;
-
-      case 'schema-primitive:URL':
-        processValue(field, 'URL', 'U', true);
-        break;
-
-      case 'schema-primitive:Number':
-        processValue(field, 'double', 'N', false);
-        break;
-
-      case 'schema-primitive:Boolean':
-        processValue(field, 'bool', 'B', false);
-        break;
-
-      default:
-        console.error(`Schema type for field '${field}' is not yet supported:`);
-        console.dir(descriptor, {depth: null});
-        process.exit(1);
-    }
+import minimist from 'minimist';
+import {Schema2Cpp} from './schema2cpp.js';
+import {Schema2Kotlin} from './schema2kotlin.js';
+
+// TODO: schemas with multiple names and schemas with parents
+// TODO: schemas with no fields
+// TODO: schemas with no names?  i.e. inline '* {Text s}'
+
+const opts = minimist(process.argv.slice(2), {
+  string: ['outdir'],
+  boolean: ['cpp', 'kotlin', 'update', 'help'],
```

Comment:
Nice. I like this centralized, mulit-lang CLI

Reply:
:+1: